src/System.Text.RegularExpressions/src/System/Text/RegularExpressions/Regex.Cache.cs

```diff
@@ -114,17 +140,17 @@ public override bool Equals(object obj)
 
             public bool Equals(CachedCodeEntryKey other)
             {
-                return this == other;
+                return _pattern.Equals(other._pattern) && _options == other._options && _cultureKey.Equals(other._cultureKey);
```

Comment:
why are putting the logic from == to Equals? Just trying to understand the change. And why are you changing the string comparison from == to Equals? (cultureKey and pattern)

Reply:
I suggested this, because Dictionary will go through Equals, rather than ==. May have no effect, may have a slight positive effect.

Reply:
Shouldn't matter as the method callees are being inlined here.